### DISCUSSION TITLE
request-transformer-advanced - Check Header is blank before transforming it to querystring

### DISCUSSION THREAD

**Discussion by slrwin** [Kong/kong#11863] [Help]
Hi Team,

Good day!!

We are using Kong request-transformer-advanced plugin to take Headers and send them as query params. Like in below snippet.

With this approach plugin is always sending the query params even though the values are blank. Please advise how can we skip query param if the headr is empty or not exists.

`x-kong-plugin-request-transformer-advanced:
  config:
    add:
      querystring:
      - "$select:$(headers[\"select\"])"
      - "$filter:$(headers[\"filter\"])"
      - "$orderby:$(headers[\"orderby\"])"
      - "$top:$(headers[\"top\"])"
      - "$count:$(headers[\"count\"])"
      - "$skiptoken:$(headers[\"skiptoken\"])"
      - "$expand:$(headers[\"expand\"])"
      - "$apply:$(headers[\"apply\"])"`

Thanks in Advance!!

**Comment by StarlightIbuki** [comment]
Hi. `request-transformer-advanced` is an enterprise feature. You could reach out to customer support for help.
